hyperspy/_signals/eels.py

```diff
@@ -1433,6 +1487,197 @@ def create_model(self, ll=None, auto_background=True, auto_add_edges=True,
                           dictionary=dictionary)
         return model
 
+    def plot(self, include_edges=False, only_edges=('Major', 'Minor'), 
+             **kwargs):
+        """Plot the EELS spectrum. Markers indicating the position of the 
+        EELS edges can be added.
+        
+        Parameters
+        ----------
+        include_edges : {False, True, list of string or string}
+            If True, draws on s.metadata.Sample.elements for edges.
+            Alternatively, provide a string of a single edge, or an iterable 
+            containing a list of valid elements, EELS families or edges. For
+            example, an element should be 'Zr', an element edge family should 
+            be 'Zr_L' or an EELS edge 'Zr_L3'.
+        only_edges : tuple of string
+            Either 'Major' or 'Minor'. Defaults to both.
+        kwargs
+            The extra keyword arguments for plot()
+        """
+        
+        super().plot(**kwargs)
+
+        if include_edges is not False:
+            edges = self._get_edges_to_plot(include_edges, only_edges)
+            self.put_label_on_signal(edges)
+
+    def put_label_on_signal(self, edges, vertical_line_marker=None, 
+                            text_marker=None):
+        """Put the EELS edge label (vertical line segment and text box) on 
+        the signal 
+
+        Parameters
+        ----------
+        edges : iterable
+            A sequence of strings contains edges in the format of 
+            element_subshell for EELS. Could be a dictionary specifying the 
+            energy value or just strings. For example, ['Fe_L2', 'O_K'] or 
+            {'Fe_L2': 721.0, 'O_K': 532.0}
+        vertical_line_marker :  list
+            A list contains HyperSpy's vertical line segment marker, if None,
+            determine from the given edges
+        text_marker :  list
+            A list contains HyperSpy's text box marker, if None,
+            determine from the given edges
+
+        Raises
+        ------
+        ValueError
+            If the size of edges, vertical_line_marker and text_marker do not
+            match.
+        """
+        
+        if vertical_line_marker is None or text_marker is None:
+            # get position of markers for edges if no marker is provided
+            slp = SpectrumLabelPosition(self)
+            vertical_line_marker, text_marker = slp.get_markers(edges)
+        if len(vertical_line_marker) != len(text_marker) or \
+            len(edges) != len(vertical_line_marker):
+            raise ValueError('The size of edges, vertical_line_marker and '
+                             'text_marker needs to be the same.')
+        
+        # add the markers to the signal and store them
+        self.add_marker(vertical_line_marker + text_marker)
+        info = dict(zip(edges, zip(vertical_line_marker, text_marker)))
+        self._edge_markers = {k:list(v) for k, v in info.items()}
```

Comment:
Is the `info` dictionary necessary? Alternatively, do the dictionary values need to be a list and not a tuple?
```suggestion
        self._edge_markers = {k:list(v) for k, v in zip(edges, zip(vertical_line_marker, text_marker))}
```

Reply:
Used `map` to change the zip's tuple to list in one go now. The values of the dictionary (which stores the markers) needs to be a list to allow popping and removing of the markers.